addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -520,41 +617,34 @@ private synchronized boolean sendCubeCommand(CubeCommand command) {
      * @param terminator String with ending messagetype e.g. L:
      * @throws IOException
      */
-    private void readliness(String terminator) throws IOException {
-        if (terminator == null) {
-            return;
-        }
-        boolean cont = true;
-        while (cont) {
+    private void readLines(String terminator) throws IOException {
+        while (true) {
```

Comment:
There are various return statements in the block. The previous logic was, subjectively, harder to read.